doc/src/vpr/fasm.rst

```diff
@@ -0,0 +1,264 @@
+     </meta>
+   </metadata>
+
+When specifying a FASM features with more than one bit, explicitly specify the
+bit range being set.  This is required because "genfasm" does not have access
+to the actual bit database, and would otherwise not have the width of the
+feature.
+
+When "fasm_type" is "SPLIT_LUT", "fasm_lut" must specify both the feature that
+represents the LUT table storage features and the pb_type path to the LUT
+being specified.  Example:
+
+.. code-block:: xml
+
+   <metadata>
+     <meta name="fasm_type">SPLIT_LUT</meta>
+     <meta name="fasm_lut">
+       ALUT.INIT[31:0] = BEL_LT-A5LUT[0]
+       ALUT.INIT[63:32] = BEL_LT-A5LUT[1]
+     </meta>
+   </metadata>
+
+In this case, the LUT in pb_type BEL_LT-A5LUT[0] will use INIT[31:0], and the
+LUT in pb_type BEL_LT-A5LUT[1] will use INIT[63:32].
+
+Within tile interconnect features
+---------------------------------
+
+When a tile has interconnect feature, e.g. output muxes, the "fasm_mux" tag
+should be attached to the interconnect tag, likely the ``<direct>`` or
+``<mux>`` tags.  From the perspective of genfasm, the ``<direct>`` and
+``<mux>`` tags are equivalent.  The syntax for the "fasm_mux" newline
+separated relationship between mux input wire names and FASM features.
+Example:
+
+.. code-block:: xml
+
+    <mux name="D5FFMUX" input="BLK_IG-COMMON_SLICE.DX BLK_IG-COMMON_SLICE.DO5" output="BLK_BB-SLICE_FF.D5[3]" >
+      <metadata>
+        <meta name="fasm_mux">
+          BLK_IG-COMMON_SLICE.DO5 : D5FFMUX.IN_A
+          BLK_IG-COMMON_SLICE.DX : D5FFMUX.IN_B
+        </meta>
+      </metadata>
+    </mux>
+
+The above mux connects input BLK_IG-COMMON_SLICE.DX or BLK_IG-COMMON_SLICE.DO5
+to BLK_BB-SLICE_FF.D5[3].  When VPR selects BLK_IG-COMMON_SLICE.DO5 for the
+mux, "genfasm" will emit D5FFMUX.IN_A, etc.
+
+There is not a requirement that all inputs result in a feature being set.
+In cases where some mux selections result in no feature being set, use "NULL"
+as the feature name.  Example:
+
+.. code-block:: xml
+
+    <mux name="CARRY_DI3" input="BLK_IG-COMMON_SLICE.DO5 BLK_IG-COMMON_SLICE.DX" output="BEL_BB-CARRY[2].DI" >
+      <metadata>
+        <meta name="fasm_mux">
+          BLK_IG-COMMON_SLICE.DO5 : CARRY4.DCY0
+          BLK_IG-COMMON_SLICE.DX : NULL
+        </meta>
+      </metadata>
+    </mux>
+
+The above examples all used the ``<mux>`` tag.  The "fasm_mux" metadata key
+can also be used with the ``<direct>`` tag in the same way, example:
+
+.. code-block:: xml
+
+    <direct name="WA7"  input="BLK_IG-SLICEM.CX" output="BLK_IG-SLICEM_MODES.WA7">
+      <metadata>
+        <meta name="fasm_mux">
+          BLK_IG-SLICEM.CX = WA7USED
+        </meta>
+      </metadata>
+    </direct>
+
+Passing parameters through to the FASM Output
+---------------------------------------------
+
+In many cases there are parameters that need to be passed directly from the
+eblif input to the FASM file.  These can be passed into a FASM feature via the
```

Comment:
Optionally could add a cross-ref to elibf documentation: https://docs.verilogtorouting.org/en/latest/vpr/file_formats/#extended-blif-eblif

Reply:
Sure